tt_metal/detail/tt_metal.hpp

```diff
@@ -485,5 +485,21 @@ namespace tt::tt_metal{
                 while(not target_device->work_executor.worker_queue.empty());
             }
         }
+        
+        // TODO: why is this in program.hpp
+        template <typename CoreRangeContainer>
+        vector<pair<transfer_info_cores, uint32_t>> extract_dst_noc_multicast_info(Device* device, const CoreRangeContainer& ranges, const CoreType core_type) {
```

Comment:
I suppose this was in program.hpp originally because it the return type includes `transfer_info_cores` which is defined in `impl/program/program_device_map.hpp`. All that is is a `std::variant<CoreCoord, CoreRange>` so maybe we can pull both out. As to where, I'd imagine it should live alongside other noc helper functions, but I'm not sure if we have any - maybe @aliuTT knows?

Reply:
There is probably a fixed set of CoreRangeContainer types. We can use explicit instantiation

Reply:
is this a good enough temporary home until we find something better? @tt-dma @pgkeller